### PR TITLE
Add a returnable cancellable implementation

### PR DESCRIPTION
**Pull request by adrwa**

### CODE REVIEW

File: src/main/java/org/sandboxpowered/eventhandler/BooleanFunction.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.sandboxpowered.eventhandler;
+
+@FunctionalInterface
+public interface BooleanFunction<T> {
```
Comment (UpcraftLP) [inline src/main/java/org/sandboxpowered/eventhandler/BooleanFunction.java]: i.e. a `Predicate`

**Review by UpcraftLP — CHANGES_REQUESTED** [review]

File: src/main/java/org/sandboxpowered/eventhandler/Cancellable.java [hunk 2]
```diff
@@ -2,20 +2,15 @@
...
 
-public class Cancellable implements BooleanSupplier {
-    static final BooleanSupplier ALWAYS_FALSE = () -> false;
-    private boolean cancelled;
-
-    public boolean isCancelled() {
-        return cancelled;
-    }
+public interface Cancellable extends BooleanSupplier {
+    BooleanSupplier ALWAYS_FALSE = () -> false;
```
Comment (MsRandom) [inline src/main/java/org/sandboxpowered/eventhandler/Cancellable.java]: Isn't this field no longer used?

**Review by MsRandom — CHANGES_REQUESTED** [review]

File: src/main/java/org/sandboxpowered/eventhandler/EventHandler.java [hunk 3]
```diff
@@ -5,19 +5,22 @@
 import java.util.function.Consumer;
 
 public interface EventHandler<T> {
```
Comment (LemmaEOF) [inline src/main/java/org/sandboxpowered/eventhandler/EventHandler.java:8]: `returnable` should probably be `postReturnable` and `cancellable` should probably be `postCancellable`

**Review by LemmaEOF — CHANGES_REQUESTED** [review]

File: src/main/java/org/sandboxpowered/eventhandler/BooleanFunction.java (hunk 1 above)
Comment (adrwa) [reply src/main/java/org/sandboxpowered/eventhandler/BooleanFunction.java]: i always forget which java function interface is which

**Review by MsRandom — APPROVED** [review]

**Review by LemmaEOF — APPROVED** [review]

**Review by UpcraftLP — APPROVED** [review]

**Pull request merged**